compiler/testData/codegen/bytecodeListing/coroutines/tcoContinuation_ir.txt

```diff
@@ -301,36 +257,80 @@ public final class TcoContinuationKt$transform$$inlined$flow$3$1 {
+public final class TcoContinuationKt$transform$$inlined$flow$3 {
+    synthetic final field $this$inlined: Flow
+    synthetic final field $transformer$inlined: kotlin.jvm.functions.Function3
+    inner class TcoContinuationKt$transform$$inlined$flow$3
+    inner class TcoContinuationKt$transform$$inlined$flow$3$1
+    public method <init>(p0: Flow, p1: kotlin.jvm.functions.Function3): void
+    public @org.jetbrains.annotations.Nullable method collect$$forInline(@org.jetbrains.annotations.NotNull p0: FlowCollector, @org.jetbrains.annotations.NotNull p1: kotlin.coroutines.Continuation): java.lang.Object
+    public @org.jetbrains.annotations.Nullable method collect(@org.jetbrains.annotations.NotNull p0: FlowCollector, @org.jetbrains.annotations.NotNull p1: kotlin.coroutines.Continuation): java.lang.Object
+}
+
+@kotlin.Metadata
+public final class TcoContinuationKt$transform$lambda-1$$inlined$collect$1$1 {
+    field label: int
+    synthetic field result: java.lang.Object
+    synthetic final field this$0: TcoContinuationKt$transform$lambda-1$$inlined$collect$1
+    inner class TcoContinuationKt$transform$lambda-1$$inlined$collect$1
+    inner class TcoContinuationKt$transform$lambda-1$$inlined$collect$1$1
+    public method <init>(p0: TcoContinuationKt$transform$lambda-1$$inlined$collect$1, p1: kotlin.coroutines.Continuation): void
+    public final @org.jetbrains.annotations.Nullable method invokeSuspend(@org.jetbrains.annotations.NotNull p0: java.lang.Object): java.lang.Object
+}
+
+@kotlin.Metadata
+public final class TcoContinuationKt$transform$lambda-1$$inlined$collect$1 {
+    synthetic final field $this$inlined: FlowCollector
+    synthetic final field $transformer$inlined: kotlin.jvm.functions.Function3
+    inner class TcoContinuationKt$transform$lambda-1$$inlined$collect$1
+    inner class TcoContinuationKt$transform$lambda-1$$inlined$collect$1$1
+    public method <init>(p0: kotlin.jvm.functions.Function3, p1: FlowCollector): void
+    public @org.jetbrains.annotations.Nullable method emit$$forInline(p0: java.lang.Object, @org.jetbrains.annotations.NotNull p1: kotlin.coroutines.Continuation): java.lang.Object
+    public @org.jetbrains.annotations.Nullable method emit(p0: java.lang.Object, @org.jetbrains.annotations.NotNull p1: kotlin.coroutines.Continuation): java.lang.Object
+}
+
+@kotlin.Metadata
+public final class TcoContinuationKt$transform$lambda-1$$inlined$collect$2$1 {
     field label: int
     synthetic field result: java.lang.Object
-    synthetic final field this$0: TcoContinuationKt$transform$$inlined$flow$3$lambda$1
-    inner class TcoContinuationKt$transform$$inlined$flow$3$lambda$1
-    inner class TcoContinuationKt$transform$$inlined$flow$3$lambda$1$1
-    public method <init>(p0: TcoContinuationKt$transform$$inlined$flow$3$lambda$1, p1: kotlin.coroutines.Continuation): void
+    synthetic final field this$0: TcoContinuationKt$transform$lambda-1$$inlined$collect$2
```

Comment:
@ilmirus Please take a look.

I'm also confused a bit with `-1` in class names

Reply:
Could we switch `-` to '_' or smth else (e.g. '$')?

Reply:
LGTM, the `-1` part is needed, when there are several lambda arguments. @pyos, can you, please, add a test with something like
```
interface FlowCollector<T> {
    suspend fun emit(value: T)
}
interface Flow<T> {
    suspend fun collect(collector: FlowCollector<T>)
}
public inline fun <T> flow(crossinline block: suspend FlowCollector<T>.() -> Unit) = object : Flow<T> {
    override suspend fun collect(collector: FlowCollector<T>) = collector.block()
}
suspend inline fun <T> Flow<T>.collect(crossinline action: suspend (T) -> Unit, crossinline action2: suspend (T) -> Unit): Unit =
    collect(object : FlowCollector<T> {
        override suspend fun emit(value: T) = action(value)
    })
public inline fun <T, R> Flow<T>.transform(crossinline transformer: suspend FlowCollector<R>.(value: T) -> Unit): Flow<R> {
    return flow {
        return@flow collect({ value ->
            return@collect transformer(value)
        }, {})
    }
}
public inline fun <T, R> Flow<T>.map(crossinline transformer: suspend (value: T) -> R): Flow<R> = transform { value -> return@transform emit(transformer(value)) }
suspend fun foo() {
    flow<Int> {
        emit(1)
    }.map { it + 1 }
        .collect ({
        }, {})
}
```

Reply:
Sure, but could you please clarify what this test should check that's not covered by others?

Reply:
Actually, the `-1` part is not really necessary. Note that all of these are regenerated objects: if the names of lambdas clash, they will simply get different indices because of 07bde889b4fa7b40bb7e3b8737093ef18713eb2d

Reply:
>Could we switch - to '_' or smth else (e.g. '$')?

To answer this question, yes, `ClassCodegen.getRegeneratedObjectNameGenerator` can be changed to give these objects different names.

Reply:
I see, this should already be covered by `crossinlineSuspendLambdaInsideCrossinlineSuspendLambda.kt`.

Reply:
> To answer this question, yes, ClassCodegen.getRegeneratedObjectNameGenerator can be changed to give these objects different names

Cool, it could be postponed to separate pr